app/controllers/api/base_controller/generic.rb

```diff
@@ -253,9 +251,9 @@ def model_ident(model, type = nil)
         end
       end
 
-      def validate_id(id, type, klass)
-        return if collection_config.resource_identifier(type) != "id"
-        raise NotFoundError, "Invalid #{klass} id #{id} specified" unless id.kind_of?(Integer) || id =~ /\A\d+\z/
+      def validate_id(id, key_id, klass)
+        return if key_id != "id"
+        raise NotFoundError, "Invalid #{klass} id #{id} specified" unless id.kind_of?(Integer) || id.to_s =~ /\A\d+\z/
```

Comment:
if id is nil, then this raises ruby warnings. `to_s` removes the warning and makes it always fail (which is what we want)

Reply:
No need to check for kind_of?(Integer) anymore, then

Reply:
I was using to_s to fix the `nil`, but I guess it fixes the integers as well.

Reply:
thanks for the pointer to `integer?`. Didn't realize rails overrode that for 'integer` and `string`.